### PR TITLE
Added skip props flag

### PR DESCRIPTION
**Pull request by maslowalex**

### CODE REVIEW

**Review by cognivore — APPROVED** [review]
LGTM! If you have tested the changes, feel free to merge!

File: lib/type_class.ex [hunk 1]
```diff
@@ -252,29 +252,31 @@ defmodule TypeClass do
 
       cond do
         unquote(class).__force_type_class__() ->
-          IO.warn("""
-          The type class #{unquote(class)} has been forced to bypass \
-          all property checks for all data types. This is very rarely valid, \
-          as all type classes should have properties associted with them.
+          # IO.warn("""
```
Comment (cognivore) [inline lib/type_class.ex]: I think that this should be another configuration option + ENV option.

**Review by cognivore — CHANGES_REQUESTED** [review]

**Comment by cognivore** [comment]
https://github.com/witchcrafters/type_class/commit/ddd369949654e413a948be1d4ed1fc4b259f2f7c

Just checking in that you've added logic like this.

**Comment by maslowalex** [comment]
> 

Should we add call to `Application` env too? Like with `force_type_class`?

File: lib/type_class.ex [hunk 2]
```diff
@@ -141,22 +141,21 @@ defmodule TypeClass do
...
-            _ -> true
-          end
-          low_prio = Application.compile_env(:doma_type_class, :force_type_class, true) # Friendship ended with false, now true is my best friend
-          if is_nil(high_prio) do
-            low_prio
-          else
-            high_prio
-          end
-        end
+				# @doc "Force type classes by default, but accept configuration options based on FORCE_TYPE_CLASS env variable and :doma_type_class > :force_type_class configuration path."
```
Comment (cognivore) [inline lib/type_class.ex]: Something happened to the indentation here.

File: lib/type_class.ex [hunk 3]
```diff
@@ -263,32 +263,44 @@ defmodule TypeClass do
         def __force_type_instance__, do: @force_type_instance
       end
 
+      print_warnings = case System.get_env("PRINT_WARNINGS") do
+        nil -> nil
+        "false" -> false
+        _ -> true
```
Comment (cognivore) [inline lib/type_class.ex:271]: You have to make sure that your changes agree with each other.

The new default behaviour is to skip compile time tests (FORCE_TYPE_CLASS = true is default).

To match this new behaviour, you have to either:

 - Default PRINT_WARNINGS to false.
 - Rename PRINT_WARNINGS to NOWARN_FORCED_TYPE_CLASS and default it to true (symmetric with the FORCE_TYPE_CLASS).

File: lib/type_class.ex [hunk 4]
```diff
@@ -263,32 +263,44 @@ defmodule TypeClass do
...
         unquote(class).__force_type_class__() ->
-          # IO.warn("""
-          # The type class #{unquote(class)} has been forced to bypass \
-          # all property checks for all data types. This is very rarely valid, \
-          # as all type classes should have properties associted with them.
-
-          # For more, please see the TypeClass README:
-          # https://github.com/expede/type_class/blob/master/README.md
-          # """)
+          if print_warnings do
```
Comment (cognivore) [inline lib/type_class.ex]: Would be cool to have this as a configuration option too, just as the FORCE_TYPE_CLASS is.

**Review by cognivore — CHANGES_REQUESTED** [review]

**Review by cognivore — CHANGES_REQUESTED** [review]
- Match the semantics of these changes with the `FORCE_TYPE_CLASS`
 - Add a low_priority configuration option here, analogously with `FORCE_TYPE_CLASS`
 - Fix indentation in the `force_type_class_const` computation.

**Review by cognivore — APPROVED** [review]

**Pull request merged**